corehq/apps/events/tasks.py

```diff
@@ -40,7 +40,31 @@ def sync_mobile_worker_attendees(domain_name, user_id):
         reopen_cases(closed_cases)
 
 
+@task(queue='background_queue', ignore_result=True)
+def close_mobile_worker_attendee_cases(domain_name):
+    """
+    Close attendee cases associated with mobile workers
+    """
+    with CriticalSection(['close_mobile_worker_attendees_' + domain_name]):
+        case_changes = []
+        user_id_case_mapping = get_existing_cases_by_user_ids(domain_name)
+        mobile_worker_user_ids = {user.user_id for user in CommCareUser.by_domain(domain_name)}
+
+        for commcare_user_id, case in user_id_case_mapping.items():
+            if commcare_user_id in mobile_worker_user_ids:
+                case_changes.append((case.case_id, {}, True))
+
+        if case_changes:
+            bulk_update_cases(
+                domain_name,
+                case_changes,
+                device_id='corehq.apps.events.tasks.close_mobile_worker_attendee_cases'
+            )
+
 def get_existing_cases_by_user_ids(domain):
+    """Returns a mapping like `{user_id1: case1, user_id2: case2}`. The user_ids's are fetched from the case's
```

Comment:
@SmittieC 
The function name is a bit confusing for two reasons:
- I was expecting it to fetch cases by user id, but we give it a domain.
- I don't know what cases it will return

How about renaming the function to something like `get_user_attendee_cases_on_domain`?

Reply:
Yeah that's a good name